src/openrct2/interface/viewport.c

```diff
@@ -1236,21 +1221,17 @@ static bool sub_679074(rct_drawpixelinfo *dpi, sint32 imageId, sint16 x, sint16
         if (xEndPoint <= 0) {
             return false;
         }
-
-        if (!(image->flags & G1_FLAG_RLE_COMPRESSION)) {
-            _unk9ABDAE += x;
-        }
     }
 
     if (image->flags & G1_FLAG_RLE_COMPRESSION) {
         return sub_67933B_679788_679C4A_67A117(image->offset, xStartPoint, yStartPoint, round);
     }
 
     uint8 *offset = image->offset + (yStartPoint * image->width) + xStartPoint;
-    uint32 ebx = gUnkEDF81C;
+    uint32 imageType = gCurrentImageType;
 
-    if (!(image->flags & 2)) {
-        return sub_679236_679662_679B0D_679FF1(ebx, image, offset);
+    if (!(image->flags & G1_FLAG_1)) {
+        return sub_679236_679662_679B0D_679FF1(imageType, image, offset);
     }
 
     // The code below is untested.
```

Comment:
Please add an assert to this code. I don't think this code is needed and can be removed but best run it with an assert for a while before removal. I think we have the same in the drawing code.